wasm/demo/snippets/mandelbrot.py

```diff
@@ -3,36 +3,39 @@
 w = 50.0
 h = 50.0
 
-# to make up for the lack of `global`
-_y = {'y': 0.0}
-
-def mandel(_time_elapsed=None):
-    y = _y['y']
-    if y >= h:
-        return
-    x = 0.0
-    while x < w:
-        Zr, Zi, Tr, Ti = 0.0, 0.0, 0.0, 0.0
-        Cr = 2 * x / w - 1.5
-        Ci = 2 * y / h - 1.0
-
-        i = 0
-        while i < 50 and Tr + Ti <= 4:
-            Zi = 2 * Zr * Zi + Ci
-            Zr = Tr - Ti + Cr
-            Tr = Zr * Zr
-            Ti = Zi * Zi
-            i += 1
-
-        if Tr + Ti <= 4:
-            print('*', end='')
-        else:
-            print('·', end='')
-
-        x += 1
-
-    print()
-    _y['y'] += 1
-    request_animation_frame(mandel)
-
-request_animation_frame(mandel)
+def mandel():
+    """Print a mandelbrot fractal to the console, yielding after each character
+    is printed"""
+    y = 0.0
+    while y < h:
+        x = 0.0
+        while x < w:
+            Zr, Zi, Tr, Ti = 0.0, 0.0, 0.0, 0.0
+            Cr = 2 * x / w - 1.5
+            Ci = 2 * y / h - 1.0
+
+            i = 0
+            while i < 50 and Tr + Ti <= 4:
+                Zi = 2 * Zr * Zi + Ci
+                Zr = Tr - Ti + Cr
+                Tr = Zr * Zr
+                Ti = Zi * Zi
+                i += 1
+
+            if Tr + Ti <= 4:
+                print('*', end='')
+            else:
+                print('·', end='')
+
+            x += 1
+            yield
+
+        print()
+        y += 1
+        yield
+
+gen = mandel()
```

Comment:
Could we make this snippet such that it also works with cpython and rustpython on host? Maybe do a check on import of request_animation_frame. I wanted to use mandelbrot on PC, but it failed.

Reply:
I think I fixed it, it segfaults for me but I think that's just because I'm testing it on my phone.

Reply:
Oh, it might have been a stack overflow. Fixed now, anyway.